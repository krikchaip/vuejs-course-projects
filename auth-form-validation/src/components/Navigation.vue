<template>
  <div class="navigation">
    <router-link class="navigation__brand" to="/">
      Vue - Complete Guide
    </router-link>
    <div class="navigation__group">
      <slot></slot>
    </div>
  </div>
</template>
