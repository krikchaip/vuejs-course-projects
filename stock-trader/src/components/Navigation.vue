<template>
  <nav class="navbar navbar-expand navbar-light bg-light shadow-sm">
    <router-link
      class="navbar-brand"
      to="/">
      Stock Trader
    </router-link>
    <div class="navbar-nav mr-auto">
      <router-link
        active-class="active"
        class="nav-item nav-link"
        to="/portfolio">
        Portfolio
      </router-link>
      <router-link
        active-class="active"
        class="nav-item nav-link"
        to="/stocks">
        Stocks
      </router-link>
    </div>
    <div class="navbar-nav">
      <strong class="nav-item navbar-text">
        Funds: ${{ funds.toLocaleString() }}
      </strong>
    </div>
  </nav>
</template>

<script>
  export default {
    props: {
      funds: { type: Number, default: 0 }
    }
  }
</script>

<style scoped>
  .nav-item {
    font-size: 0.95rem;
  }
</style>
