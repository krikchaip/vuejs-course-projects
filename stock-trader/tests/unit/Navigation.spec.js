import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Navigation from '@/components/Navigation'

describe('RouterLink', () => {
  const wrapper = shallowMount(Navigation, {
    stubs: { RouterLink: RouterLinkStub }
  })
  const links = wrapper.findAll(RouterLinkStub)

  it('has only one link to "/"', () => {
    const toRoot = links.filter(w => w.props().to === '/')
    expect(toRoot.length).toBe(1)
  })

  it('has only one link to "/portfolio with custom active class"', () => {
    const toPortfolio = links.filter(w => w.props().to === '/portfolio')
    expect(toPortfolio.length).toBe(1)
    expect(toPortfolio.at(0).props().activeClass).toBeTruthy()
  })

  it('has only one link to "/stocks with custom active class"', () => {
    const toStocks = links.filter(w => w.props().to === '/stocks')
    expect(toStocks.length).toBe(1)
    expect(toStocks.at(0).props().activeClass).toBeTruthy()
  })
})

describe('Funds', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Navigation, {
      stubs: ['router-link']
    })
  })

  it('recieves via "funds" prop', () => {
    wrapper.setProps({ funds: 100 })
    expect(wrapper.props().funds).toBe(100)
  })

  it('should be a number, default is 0', () => {
    const spec = { type: Number, default: 0 }
    expect(wrapper.vm.$options.props.funds).toEqual(spec)
  })

  it('displays correctly', () => {
    wrapper.setProps({ funds: 10000 })
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('Actions', () => {
  it('uses default slot', () => {
    const Action = { name: 'Action', template: `<div/>` }
    const wrapper = shallowMount(Navigation, {
      stubs: ['router-link'],
      slots: { default: Action }
    })

    expect(wrapper.find(Action).exists()).toBeTruthy()
  })
})
