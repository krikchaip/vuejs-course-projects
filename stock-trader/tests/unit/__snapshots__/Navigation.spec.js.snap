// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funds displays correctly 1`] = `
<nav class="navbar navbar-expand navbar-light bg-light shadow-sm">
  <router-link-stub to="/" class="navbar-brand">
    Stock Trader
  </router-link-stub>
  <div class="navbar-nav mr-auto">
    <router-link-stub active-class="active" to="/portfolio" class="nav-item nav-link">
      Portfolio
    </router-link-stub>
    <router-link-stub active-class="active" to="/stocks" class="nav-item nav-link">
      Stocks
    </router-link-stub>
  </div>
  <div class="navbar-nav"> <strong class="nav-item navbar-text">
      Funds: $10,000
    </strong></div>
</nav>
`;
